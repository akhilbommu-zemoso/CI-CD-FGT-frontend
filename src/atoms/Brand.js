import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import { Link} from 'react-router-dom'
const useStyles = makeStyles((theme) => ({
    title: {
        height: '30px',
        fill: 'white',
        borderRadius: "50%",
        padding: '5px',
        marginBottom: '8px',
        marginRight: '10px',
        fontSize: '28px',
        fontWeight: 900,
        color:'#485460',
    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
}));
export default function Brand(props) {
    const classes = useStyles();
    return (
        <div>
            <List component="nav" aria-label="main mailbox folders">
                <Link className={classes.linksclass} to="/dashboard/home">
                    <div className={classes.title} id="brand">
                        {props.name}
                     </div>
                </Link>
            </List>
        </div>
    );
}
