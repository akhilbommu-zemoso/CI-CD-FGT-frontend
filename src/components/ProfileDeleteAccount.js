import React, { useState } from "react";
import { makeStyles, withStyles, fade } from "@material-ui/core/styles";
import { Link, Redirect } from 'react-router-dom'
import { logout } from "../utils/index"
const useStyles = makeStyles({
    textContent: {
        fontSize: '14px',
        color: "rgb(72,84,96)",
        marginTop: '10px',
    },
    deleteHeading: {
        fontSize: '18px',
        fontWeight: 900,
        color: "rgb(22,101,216)",
        marginBottom: "20px",
        borderBottom: "2px dashed rgb(22,101,216)",
        '   &:focus': {
            color: "red",
            borderBottom: "2px dashed red",
        },
    },
    linksclass: {
        color: "rgb(22,101,216)",
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: '#CC0000',
            textDecoration: 'none',
            borderBottom: "2px dashed #CC0000",
        },
    },
});
function ProfileDeleteAccount() {
    const [customError, setcustomError] = useState(null);
    const handleDelete = () => {
        var flag = window.confirm("You'll be deleted permanently, continue?");
        if (flag)
            postToApi();
    }
    const postToApi = () => {
        var accessToken = localStorage.getItem("accessToken");
        fetch(process.env.REACT_APP_BACKEND_USER, {
            method: 'DELETE',
            headers: {
                'Authorization': 'Bearer ' + accessToken,
            },
        }).then(response => {
            console.log("response---", response)
            if (!response.ok) {
                console.log("status--------" + response.status);
                throw new Error(response.status)
            }
            logout();
            window.location.href = window.location.origin;
        })
            .catch((error) => {
                console.error('Error Occured at:', error, typeof (error));
                if (error.message == "401") {
                    setcustomError("Unauthorized Access");
                }
                else if (error.message == "500") {
                    setcustomError("Internal Server Error");
                }
                else {
                    setcustomError("Unable to fetch resource")
                }
                return <Redirect to="/error" />
            });
    };
    const classes = useStyles();
    return (
        <div>
            <div>
                <span className={classes.deleteHeading} >
                    <Link
                        data-testid="link"
                        className={classes.linksclass}
                        onClick={() => handleDelete()}
                    >Delete my account permanently</Link>
                </span>
            </div>
            <div className={classes.textContent} data-testid="content">
                You will receive email to confirm your decision<br />
             If you delete the account data will be permanently erased. so, please
             turn on sync data to keep your files safe.
          </div>
        </div>
    );
}
export default ProfileDeleteAccount