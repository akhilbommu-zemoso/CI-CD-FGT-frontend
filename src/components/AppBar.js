import React from 'react';
import { fade, makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import NotificationsIcon from '@material-ui/icons/Notifications';
import MoreIcon from '@material-ui/icons/MoreVert';
import HelpOutlineSharpIcon from '@material-ui/icons/HelpOutlineSharp';
import ProfileList from './ProfileList'
import Brand from '../atoms/Brand'
import ProfilePicture from './ProfilePicture';
const useStyles = makeStyles((theme) => ({
    grow: {
        maxWidth: `calc(100%-240px)`,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        marginTop: '10px',
        fontWeight: 700,
        [theme.breakpoints.up('sm')]: {
            display: 'block',
        },
    },
    search: {
        position: 'relative',
        borderRadius: theme.shape.borderRadius,
        border: '1px solid #dddddd',
        backgroundColor: fade(theme.palette.common.white, 0.15),
        '&:hover': {
            backgroundColor: fade(theme.palette.common.white, 0.25),
        },
        marginRight: theme.spacing(2),
        marginLeft: 0,
        width: '100%',
        [theme.breakpoints.up('sm')]: {
            marginLeft: theme.spacing(3),
            width: 'auto',
        },
    },
    searchIcon: {
        padding: theme.spacing(0, 2),
        height: '100%',
        position: 'absolute',
        pointerEvents: 'none',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    },
    inputRoot: {
        color: 'inherit',
    },
    inputInput: {
        padding: theme.spacing(1, 1, 1, 0),

        paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
        transition: theme.transitions.create('width'),
        width: '100%',
        [theme.breakpoints.up('md')]: {
            width: '20ch',
        },
    },
    sectionDesktop: {
        display: 'none',
        [theme.breakpoints.up('md')]: {
            display: 'flex',
        },
    },
    sectionMobile: {
        display: 'flex',
        [theme.breakpoints.up('md')]: {
            display: 'none',
        },
    },
    red: {
        color: 'black',
        backgroundColor: 'rgb(255,255,255)'
    },
    space: {
        display: 'flex',
        justifyContent: 'space-between',
        width: '100%',
    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
    logo: {
        width: '30px',
        height: '30px',
        fill: 'white',
        backgroundColor: "rgb(4,136,255)",
        borderRadius: "50%",
        padding: '5px',
        marginBottom: '6px',
        marginRight: '10px',
    },
    padding: {
        paddingLeft: '10px',
        paddingRight: '20px',
    },
    nopadding: {
        padding: '0px',
    },
    avatar: {
        backgroundColor: 'rgba(182,32,224,0.71)',
        width: '30px',
        height: '30px',
        borderRadius: '50%',
        fontSize: '18px'
    },
    iconpadding: {
        paddingLeft: '15px',
        paddingRight: '15px',
    }


}));
export default function PrimarySearchAppBar(props, children) {
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);
    const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);
    const isMenuOpen = Boolean(anchorEl);
    const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);
    const handleProfileMenuOpen = (event) => {
        setAnchorEl(event.currentTarget);
    };
    const handleMobileMenuClose = () => {
        setMobileMoreAnchorEl(null);
    };
    const handleMenuClose = () => {
        setAnchorEl(null);
        handleMobileMenuClose();
    };
    const handleMobileMenuOpen = (event) => {
        setMobileMoreAnchorEl(event.currentTarget);
    };
    const menuId = 'primary-search-account-menu';
    const renderMenu = (
        <Menu
            data-testid='menuclose'
            MenuListProps={{ disablePadding: true }}
            anchorEl={anchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            id={menuId}
            keepMounted
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMenuOpen}
            onClose={handleMenuClose}
        >
            {/* rendering profile list when user clicks on appbar avtar */}
            {props.profilelist}
        </Menu>
    );
    const mobileMenuId = 'primary-search-account-menu-mobile';
    const renderMobileMenu = (
        <Menu
            data-testid='mobileclose'
            anchorEl={mobileMoreAnchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            id={mobileMenuId}
            keepMounted
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMobileMenuOpen}
            onClose={handleMobileMenuClose}
        >
            <MenuItem>
                <IconButton aria-label="show  new mails" color="inherit">
                    <Badge color="secondary">
                        <HelpOutlineSharpIcon />
                    </Badge>
                </IconButton>
                <p>Messages</p>
            </MenuItem>
            <MenuItem>
                <IconButton aria-label="show  new notifications" color="inherit">
                    <Badge color="secondary">
                        <NotificationsIcon />
                    </Badge>
                </IconButton>
                <p>Notifications</p>
            </MenuItem>
            <MenuItem onClick={handleProfileMenuOpen}>
                <IconButton
                    aria-label="account of current user"
                    aria-controls="primary-search-account-menu"
                    aria-haspopup="true"
                    color="inherit"
                >
                    <AccountCircle />
                </IconButton>
                <p>Profile</p>
            </MenuItem>
        </Menu>
    );
    return (
        <div className={classes.grow}>
            <AppBar position="fixed" className={classes.red}>
                <Toolbar className={classes.padding}>
                    <div className={classes.space} >
                        <div data-testid="brand">
                            {/* rendering the brand ZE-Transact */}
                            {props.brand}
                        </div>
                        <div >
                            <div className={classes.sectionDesktop}>
                                <IconButton aria-label="show 4 new mails" color="inherit" className={classes.iconpadding}>
                                    <Badge color="secondary" data-testid="help">
                                        <HelpOutlineSharpIcon />
                                    </Badge>
                                </IconButton>
                                <IconButton aria-label="show  new notifications" color="inherit" className={classes.iconpadding}>
                                    <Badge color="secondary" data-testid="notification">
                                        <NotificationsIcon />
                                    </Badge>
                                </IconButton>
                                <IconButton
                                    data-testid="iconbutton"
                                    edge="end"
                                    aria-label="account of current user"
                                    aria-controls={menuId}
                                    aria-haspopup="true"
                                    onClick={handleProfileMenuOpen}
                                    color="inherit"
                                >
                                    <div data-testid="picture">
                                        <ProfilePicture pictureStyle={classes.avatar} />
                                    </div>
                                </IconButton>
                            </div>
                        </div>
                    </div>
                    <div className={classes.sectionMobile}>
                        <IconButton
                            data-testid="mobilemenu"
                            aria-label="show more"
                            aria-controls={mobileMenuId}
                            aria-haspopup="true"
                            onClick={handleMobileMenuOpen}
                            color="inherit"
                        >
                            <MoreIcon />
                        </IconButton>
                    </div>
                </Toolbar>
            </AppBar>
            {renderMobileMenu}
            {renderMenu}
        </div>
    );
}
