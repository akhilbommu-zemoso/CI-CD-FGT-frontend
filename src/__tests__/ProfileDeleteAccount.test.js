import React, { Component } from 'react'
import { render,screen, fireEvent, getAllByText, waitFor } from '@testing-library/react'
import { renderHook, act } from '@testing-library/react-hooks'
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme'
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button'
import Adapter from 'enzyme-adapter-react-16';
import { create } from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });
import AppBar from '../components/AppBar'
import { findDOMNode } from 'react-dom';
import Brand from '../atoms/Brand'
import fetchMock from 'fetch-mock'
import { BrowserRouter as Router } from 'react-router-dom'
import ProfileDeleteAccount from '../components/ProfileDeleteAccount'
const setSelected = jest.fn();
it('should take a snapshot', () => {

    const { asFragment } = render(
        <Router>
            <ProfileDeleteAccount />
        </Router>
    )
    expect(asFragment(<Router><ProfileDeleteAccount itemName="Profile" /></Router>)).toMatchSnapshot()
});

it("handlesubmit event should work ", async () => {
    const { getByTestId } = render(<Router><ProfileDeleteAccount /></Router>);
    fireEvent.click(getByTestId("link"));
    fetchMock.mock("http://localhost:8001/user/users", 200);
    const res = await fetch("http://localhost:8001/user/users");
    expect(res.ok).toBeTruthy;
});

const setcustomError = jest.fn();
const postToApi = jest.fn();
const useStateSpy = jest.spyOn(React, 'useState')
useStateSpy.mockImplementation((init) => [init, setcustomError]);
describe("isAuthenticated", () => {
    // resets your mocks so tests don't bleed into each other
    afterEach(() => {
        fetchMock.restore();
        fetchMock.reset();
    });
    const handleDelete = jest.fn();
const flag=jest.fn(()=>true)
    it("is true on server success", async () => {
        const { getByTestId } = render(<Router><ProfileDeleteAccount /></Router>);
        fetchMock.mock("http://localhost:8080/api/user/users", {
            status: 200,
            body: { success: true }
        });
        fireEvent.click(getByTestId("link"))
        expect(handleDelete).toBeCalled
        window.confirm = jest.fn(() => false)
        const res = await fetch("http://localhost:8080/api/user/users")
        expect(res.ok).toBe(true);
    });
})
const handleDelete = jest.fn();

it("is true on server success", async () => {
    const { getByTestId,getByText } = render(<Router><ProfileDeleteAccount /></Router>);
    fetchMock.mock("http://localhost:8080/api/user/users", {
        status: 401,
        body: { success: "Invalid" }
    });
    fireEvent.click(getByTestId("link"))
 
    expect(handleDelete).toBeCalled
   
    expect(window.confirm()).toBeInTheDocument
    window.confirm = jest.fn().mockImplementation(() => true)
    const res = await fetch("http://localhost:8080/api/user/users")
    expect(res.ok).toBe(false);
});

