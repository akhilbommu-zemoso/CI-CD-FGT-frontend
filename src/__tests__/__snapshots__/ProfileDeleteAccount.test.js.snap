// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <span
        class="makeStyles-deleteHeading-2"
      >
        <a
          class="makeStyles-linksclass-3"
          data-testid="link"
          href=""
        >
          Delete my account permanently
        </a>
      </span>
    </div>
    <div
      class="makeStyles-textContent-1"
      data-testid="content"
    >
      You will receive email to confirm your decision
      <br />
      If you delete the account data will be permanently erased. so, please turn on sync data to keep your files safe.
    </div>
  </div>
</DocumentFragment>
`;
